### PR TITLE
[wip] revert compling cache-dir and remove non standard code from cache-dir

### LINKED ISSUE #6102: [v2] provide non-traspiled version of default-html.js to copy from
## Description

I want to customize `html.js` but the `default-html.js` file is transpiled to ES5 by default.

### Steps to reproduce

Install all the dependencies from `package.json` using `yarn`, run `gatsby develop`, it will generate `.cache` folder.

Copy the `default-html.js` in `.cache` folder to `src/html.js`

```batch
cp .cache/default-html.js src/html.js
```

### Expected result

The content of `default-html.js` should be ES6, not be transpiled by default.

### Actual result

The content of `html.js` is not ES6

### Environment

```batch
  System:
    OS: macOS High Sierra 10.13.5
    CPU: x64 Intel(R) Core(TM) i7-3520M CPU @ 2.90GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 8.11.1 - /usr/local/bin/node
    Yarn: 1.7.0 - /usr/local/bin/yarn
    npm: 5.6.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 67.0.3396.87
    Safari: 11.1.1
  npmPackages:
    gatsby: next => 2.0.0-beta.9
    gatsby-plugin-no-sourcemaps: next => 2.0.0-beta.2
    gatsby-plugin-react-helmet: next => 3.0.0-beta.2
    gatsby-plugin-styled-components: next => 3.0.0-beta.2
```

### File contents (if changed)

`gatsby-config.js`:
```javascript
module.exports = {
  plugins: [
    'gatsby-plugin-react-helmet',
    'gatsby-plugin-styled-components',
    'gatsby-plugin-no-sourcemaps',
  ],
}
```
`package.json`: 
```json
  "dependencies": {
    "babel-plugin-styled-components": "^1.5.1",
    "gatsby": "next",
    "gatsby-plugin-no-sourcemaps": "next",
    "gatsby-plugin-react-helmet": "next",
    "gatsby-plugin-styled-components": "next",
    "prop-types": "^15.6.2",
    "react": "^16.4.1",
    "react-dom": "^16.4.1",
    "react-feather": "^1.1.0",
    "react-helmet": "^5.2.0",
    "react-router-dom": "^4.3.1",
    "sanitize.css": "^5.0.0",
    "styled-components": "^3.3.3"
  },
  "devDependencies": {
    "babel-eslint": "^8.2.3",
    "eslint": "^4.19.1",
    "eslint-config-standard": "^11.0.0",
    "eslint-config-standard-react": "^6.0.0",
    "eslint-plugin-import": "^2.12.0",
    "eslint-plugin-node": "^6.0.1",
    "eslint-plugin-promise": "^3.8.0",
    "eslint-plugin-react": "^7.9.1",
    "eslint-plugin-standard": "^3.1.0"
  },
  "scripts": {
    "build": "gatsby build",
    "start": "gatsby develop --https"
  },
```
`gatsby-node.js`:
```javascript
const { resolve } = require('path')

exports.createPages = ({ actions }) => {
  const { createPage } = actions

  createPage({
    path: '/',
    matchPath: '/:path',
    component: resolve(__dirname, './src/app.js'),
  })
}
```
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A

### LINKED ISSUE #6103: [v2] IE 11 'Promise' is undefined error
## Description

IE 11 throws "Unhandled promise rejection ReferenceError: 'Promise' is undefined" error in v2 sites. This can be observed at next.gatsbyjs.org and can replicate it locally on my site. Including promise polyfill fixes the issue, but from looking at documentation `babel-polyfill` is meant to handle that based on defined browsers list.

### Steps to reproduce
- Open next.gatsbyjs.org in IE 11
- Observe "Unhandled promise rejection ReferenceError: 'Promise' is undefined" error thrown in console longs

### Expected result
No promise related errors.

### Actual result
```
Unhandled promise rejection ReferenceError: 'Promise' is undefined
   "Unhandled promise rejection"
   {
      [functions]: ,
      __proto__: { },
      description: "'Promise' is undefined",
      message: "'Promise' is undefined",
      name: "ReferenceError",
      number: -2146823279,
      stack: "ReferenceError: 'Promise' is undefined
   at t.components.component---src-pages-index-js (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:6:70060)
   at Anonymous function (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:6:32261)
   at O (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:24:11838)
   at S (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:6:31869)
   at E (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:6:32413)
   at A.getResourcesForPathname (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:6:34343)
   at Anonymous function (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:24:17870)
   at a (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:24:10747)
   at Anonymous function (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:24:10894)
   at c (https://next.gatsbyjs.org/app-ba276550eb4051a9bc0a.js:24:8307)"
   }
```

### Environment
This is from my own site directory.
```
  System:
    OS: macOS High Sierra 10.13.5
    CPU: x64 Intel(R) Core(TM) i5-4670 CPU @ 3.40GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 8.11.1 - /usr/local/bin/node
    Yarn: 1.6.0 - /usr/local/bin/yarn
    npm: 6.1.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 67.0.3396.87
    Firefox: 60.0.1
    Safari: 11.1.1
  npmPackages:
    gatsby: ^2.0.0-beta.9 => 2.0.0-beta.9 
    gatsby-plugin-glamor: ^2.0.0-beta.2 => 2.0.0-beta.2 
    gatsby-plugin-manifest: ^2.0.2-beta.2 => 2.0.2-beta.2 
    gatsby-plugin-netlify: ^2.0.0-beta.2 => 2.0.0-beta.2 
    gatsby-plugin-offline: ^2.0.0-beta.2 => 2.0.0-beta.2 
    gatsby-plugin-react-helmet: ^3.0.0-beta.2 => 3.0.0-beta.2 
    gatsby-plugin-sitemap: ^2.0.0-beta.2 => 2.0.0-beta.2 
  npmGlobalPackages:
    gatsby-cli: 2.0.0-beta.1
```

### File contents (if changed)

`gatsby-config.js`: N/A 
`package.json`: N/A 
`gatsby-node.js`: N/A 
`gatsby-browser.js`: N/A 
`gatsby-ssr.js`: N/A

### PR DESCRIPTION
**Pull request by pieh**
#5637 might be to aggressive to transpile it - right now `@babel/preset-env` doesn't know it needs to polyfill `Promise` used in `cache-dir`, which breaks runtime in ie 11 ( #6103 )

This attempt just removes flow types and move cache-dir to same place it was before.

closes #6103 
closes #6102

### CODE REVIEW

**Comment by pieh** [comment]
Actually, because we don't use `flow` extensively in `cache-dir`, removing flow stuff manually from cache-dir would yield same result without need for "build" step for it

**Comment by jquense** [comment]
It'd be cool if there was like an eslint plguin for only allowing stuff preset-env can compile, like nothing that isn't standardized or implemented

**Comment by pieh** [comment]
> It'd be cool if there was like an eslint plguin for only allowing stuff preset-env can compile, like nothing that isn't standardized or implemented

I didn't find anything like that for eslint, but maybe we could add test to try to run babel on cache-dir with minimal config to catch changes that would add any non standard stuff?

**Comment by pieh** [comment]
@jquense how about something like https://github.com/pieh/gatsby/commit/f300bc87f89b091e7829f19d0f232dc9be849ba1 (proof of concept)
it would fail when there would be (for example) flow types:
```
 FAIL  packages\gatsby\cache-dir\__tests__\minimal-config.js
  × Builds cache-dir with minimal config (1085ms)

  ● Builds cache-dir with minimal config

    expect(received).toEqual(expected)

    Expected value to equal:
      ""
    Received:
      "{ SyntaxError: D:\\dev\\gatsby-v2\\packages\\gatsby\\cache-dir\\prefetcher.js: Unexpected token, expected \";\" (3:5)

      1 | // @flow
      2 |
    > 3 | type PrefetcherOptions = {
        |      ^
      4 |   fetchNextResource: () => ?Promise,
      5 | }
      6 |
        at _class.raise (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:3893:15)
        at _class.unexpected (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:5222:16)
        at _class.semicolon (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:5206:40)
        at _class.parseExpressionStatement (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:7635:10)
        at _class.parseStatementContent (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:7234:19)
        at _class.parseStatement (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:7115:17)
        at _class.parseBlockOrModuleBlockBody (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:7666:23)
        at _class.parseBlockBody (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:7653:10)
        at _class.parseTopLevel (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:7080:10)
        at _class.parse (D:\\dev\\gatsby-v2\\node_modules\\@babel\\parser\\lib\\index.js:8466:17)
      pos: 17,
      loc: Position { line: 3, column: 5 },
      code: 'BABEL_PARSE_ERROR' }
    "

      26 |
      27 |   spawn.on(`close`, function () {
    > 28 |     expect(stderr).toEqual(``)
      29 |     expect(stdout).toEqual(
      30 |       expect.stringMatching(`Successfully compiled`),
      31 |     )

      at ChildProcess.<anonymous> (packages/gatsby/cache-dir/__tests__/minimal-config.js:28:20)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        2.695s
```

File: packages/gatsby/cache-dir/json-store.js [hunk 1]
```diff
@@ -4,7 +4,10 @@ import PageRenderer from "./page-renderer"
 import { StaticQueryContext } from "gatsby"
 import socketIo, { getStaticQueryData, getPageQueryData } from "./socketIo"
 
-const getPathFromProps = props => props.pageResources?.page?.path
+const getPathFromProps = props =>
```
Comment (KyleAMathews) [inline packages/gatsby/cache-dir/json-store.js]: Why revert this?

Comment (pieh) [reply packages/gatsby/cache-dir/json-store.js]: Maybe I don't need to revert this, I thought this needed something like `transform-optional-chaining` plugin. I'll re-check it.

**Pull request merged**